Source/Storytellers/Nesi/NesiStoryteller.cs

```diff
@@ -140,24 +164,32 @@ private void UpdateState()
                     return;
                 }
 
-                if (StorytellerData.state == 0 && new Random(Find.World.ConstantRandSeed).Next(1, 10) == 2)
+                if (
+                currentState == StorytellerState.Neutral &&
+                worldRandomGenerator.Next(1, 10) == 2)
                 {
-                    StorytellerData.state += new Random(Find.World.ConstantRandSeed).Next(-1, 2);
+                    StorytellerData.state += worldRandomGenerator.Next(-1, 2);
                     StorytellerData.dayLastUpdated = GenDate.DaysPassed;
                     return;
                 }
 
-                // FIXME: Weird use of | instead of ||?
                 // Currently checks if state is -1/Hunting
-                if (StorytellerData.state == (StorytellerState.Aggressive | StorytellerState.HyperAggressive) &&
-                    new Random(Find.World.ConstantRandSeed).Next(1, 10) == 2)
+                if (currentState == StorytellerState.Hunting &&
+                    worldRandomGenerator.Next(1, 10) == 2)
```

Comment:
The FIXME was intended to ask if this was the intended behavior or a mistake, since the code reads as if the intent was to make it trigger on Aggressive or HyperAggressive state, and Hunting was just a coincidence.